internal/metrics/model.go

```diff
@@ -0,0 +1,28 @@
+package metrics
+
+import "github.com/nginx/agent/v3/internal/bus"
+
+type (
+	// Represents a single entry for a data source, is one of [counter, gauge, histogram, summary].
+	DataEntry struct {
+		Name        string
+		Type        string
+		SourceType  string
+		Description string
+		Values      []DataPoint
+	}
+
+	// A single data point for an entry. An entry can have multiple points.
+	DataPoint struct {
+		Name   string
+		Labels map[string]string
+		Value  float64
+	}
```

Comment:
I'm not quite happy with having these intermediate types that we very quickly transform into OTel data in `metrics.go`. I'm looking into getting rid of these altogether, and using `metricdata.DataPoint`s directly instead, which we could then push out in `metrics.go` to the appropriate meters.

Reply:
@dhurley made the fair point that we will likely want to support exporting to other types than OTel in the future, therefore having tight coupling to OTel might not be a great idea. So we will support this internal model, and then perform transformations to the correct target types based on the target data sink.